src/components/gridView/index.tsx

```diff
@@ -177,30 +181,39 @@ class GridView extends UIComponent<GridViewProps, GridViewState> {
 
   renderItem = (item: GridListItemProps, index: number) => {
     const {itemSize} = this.state;
-    const {items, itemSpacing} = this.props;
-
+    const {items, itemSpacing, renderCustomItem} = this.props;
     const {numColumns = DEFAULT_NUM_COLUMNS} = this.state;
+
     const itemsCount = _.size(items);
     const rowCount = Math.ceil(itemsCount / numColumns);
     const isLastItemInRow = (index + 1) % numColumns === 0;
     const isLastRow = index + 1 > (rowCount - 1) * numColumns;
     const isLastItem = index === itemsCount - 1;
     const size =
-      typeof item.itemSize === 'object' ? {width: itemSize, height: item.itemSize?.height || itemSize} : itemSize;
-    return (
-      <GridListItem
-        key={index}
-        {...item}
-        itemSize={size}
-        containerStyle={[
-          !isLastItemInRow && {marginRight: itemSpacing},
-          !isLastRow && {marginBottom: itemSpacing},
-          item.containerStyle
-        ]}
-      >
-        {isLastItem && this.renderLastItemOverlay()}
-      </GridListItem>
-    );
+      typeof item.itemSize === 'object'
+        ? {
+          width: itemSize,
+          height: item.itemSize?.height || itemSize
+        }
+        : itemSize;
+
+    const itemProps = {
+      key: index,
+      ...item,
+      itemSize: size,
+      containerStyle: [
+        !isLastItemInRow && {
+          marginRight: itemSpacing
+        },
+        !isLastRow && {
+          marginBottom: itemSpacing
+        },
+        item.containerStyle
+      ],
+      children: isLastItem ? this.renderLastItemOverlay() : undefined
+    };
+
+    return renderCustomItem ? renderCustomItem(itemProps) : <GridListItem {...itemProps}/>;
```

Comment:
We don't usually send (as far as I remember) our configuration to the custom render function, is this a new approach or a special case?
Also, I feel like the name (`item` in the definition above) should be more along "suggested props" if this is how it's called, not sure how to call it though.

Reply:
Not sure I understand the comment. 
What you mean by configuration? 

What I did is just pass the props I usually pass to `GridListItem` to the custom render so it can use it. 
Technically, we render in private a wrapper of GridListItem (check the private PR)

Reply:
I totally missed the email from private (probably archived it, thinking it's the same one).